src/main/java/io/cryostat/net/web/http/api/v2/RulesPostHandler.java

```diff
@@ -127,6 +127,10 @@ public boolean isOrdered() {
             case JSON:
                 try {
                     rule = gson.fromJson(params.getBody(), Rule.class);
+
+                    if (rule == null) {
+                        throw new IllegalArgumentException("No JSON found");
```

Comment:
I think the message here could be a little more precise. Something like `POST body was null`, that indicates exactly to the user/client what their mistake was.